src/Neo/SmartContract/Native/ContractManagement.cs

```diff
@@ -145,6 +145,19 @@ public ContractState GetContract(IReadOnlyStore snapshot, UInt160 hash)
             return snapshot.TryGet(key, out var item) ? item.GetInteroperable<ContractState>(false) : null;
         }
 
+        /// <summary>
+        /// Check if exists the deployed contract with the specified hash.
+        /// </summary>
+        /// <param name="snapshot">The snapshot used to read data.</param>
+        /// <param name="hash">The hash of the deployed contract.</param>
+        /// <returns>True if deployed contract exists.</returns>
+        [ContractMethod(Hardfork.HF_Echidna, CpuFee = 1 << 14, RequiredCallFlags = CallFlags.ReadStates)]
+        public bool IsContract(IReadOnlyStore snapshot, UInt160 hash)
+        {
+            var key = CreateStorageKey(Prefix_Contract, hash);
+            return snapshot.Contains(key);
```

Comment:
i have a question, are you sure there is no such situation that you can find the contact in the state, but it can not convert to ContractState?

Reply:
https://github.com/neo-project/neo/issues/2829

Reply:
It's a contract, it will fault now or when you call it, but we need to trust in our code

Reply:
Or maybe it's irrelevant now. In general, this API is useful, it has lower overhead.

Reply:
> What's the current GetContract() result for this contract?

After https://github.com/neo-project/neo/pull/2948, the deployment transaction of the contract mentioned in #2829 is FAULTed. So now we don't have such "broken" contracts in the node's storage, which means that `GetContract()` will return `null` for such contract. 

> there is no such situation that you can find the contact in the state, but it can not convert to ContractState?

So this situation is not possible. If we have contract in the storage, then this contract can always be converted to `ContractState`.